### ISSUE TITLE
Deleting data causes shard to be dropped

### ISSUE THREAD

**Issue by daviskirk** [influxdata/influxdb#9178]
### Bug report

__System info:__ influxdb:1.3.7, running in docker persisted via host mount

__Steps to reproduce:__

This is a followup issue on https://github.com/influxdata/influxdb/issues/8937 , occurred today, approx. 1 month after upgrading to 1.3.7.

1. Write data to default database
2. Delete data in a few different series rapidly 

__Expected behavior:__

Only the deleted times and series are deleted.

__Actual behavior:__ 

* The entire shard is dropped (or at least almost all data within the limits of the shard is no longer queryable)
  `570 default  autogen          570         2017-09-25T00:00:00Z 2017-12-18T00:00:00Z 2022-12-12T00:00:00Z`  
  a few series still have data for this period (although I cannot find any other shard that corresponds to this time frame)
* After this, writes to that shard succeed and are queryable but then disappear after a while 
* Writes continue to succeed (for the other 50% of time series)
* The shard file itself was not empty (166MB)
* In the client code there were no "drop shard" commands used, `delete from` is used with specific time series (but never for exactly the shard period)
* Inside the shard directory the following files are present: 
```du -sh *
163M	000000601-000000005.tsm
3.1M	000000603-000000001.tsm
``` 

__Additional info:__ 

The relevant part of the output of `show shard groups`:

```
195 default  autogen          195         2017-01-09T00:00:00Z 2017-01-16T00:00:00Z 2022-01-10T00:00:00Z 
293 default  autogen          293         2017-01-16T00:00:00Z 2017-01-23T00:00:00Z 2022-01-17T00:00:00Z 
378 default  autogen          378         2017-01-16T00:00:00Z 2017-04-10T00:00:00Z 2022-04-04T00:00:00Z 
432 default  autogen          432         2017-04-10T00:00:00Z 2017-07-03T00:00:00Z 2022-06-27T00:00:00Z 
433 default  autogen          433         2017-07-03T00:00:00Z 2017-09-25T00:00:00Z 2022-09-19T00:00:00Z 
570 default  autogen          570         2017-09-25T00:00:00Z 2017-12-18T00:00:00Z 2022-12-12T00:00:00Z 
571 default  autogen          571         2017-12-18T00:00:00Z 2018-03-12T00:00:00Z 2023-03-06T00:00:00Z 
654 default  autogen          654         2018-03-12T00:00:00Z 2018-06-04T00:00:00Z 2023-05-29T00:00:00Z 
655 default  autogen          655         2018-06-04T00:00:00Z 2018-08-27T00:00:00Z 2023-08-21T00:00:00Z 
652 default  autogen          652         2019-10-21T00:00:00Z 2020-01-13T00:00:00Z 2025-01-06T00:00:00Z
```

**Comment by e-dard** [comment]
@daviskirk could you try upgrading to the latest release? It looks like https://github.com/influxdata/influxdb/pull/8981 might have fixed the issue. #8981 is included in the `1.4.2` release but as far as I know it wasn't backported to 1.3.7

**Comment by mmuller99** [comment]
@e-dard is the 1.4.2 release official yet? We are in the same boat, also see [#8981](https://github.com/influxdata/influxdb/issues/8981), and the official site is stil on 1.3 and there has been no feedback from the team on [#9133](https://github.com/influxdata/influxdb/issues/9133). As a matter of principal, we would only then upgrade our production systems

**Comment by max3163** [comment]
@e-dard The latest version 1.4.2 don't fix the problem of the drop shard when you delete multiple points in a raw. 
I try the latest code in 1.5 and the problem still exist too, even worse as when I delete just one point, I lost nearly all the series from the measurement.

**Comment by daviskirk** [comment]
I can second @max3163 findings.  Is there a workaround for deleting data another "safe" way?  This is obviously pretty much a show stopper for running influx anywhere remotely important.

**Comment by jwilder** [comment]
This should be fixed in master.

* #9084 reworked how deletes are handled and fixed many long-standing issues with deletes.  Some of the changes in there are related to the issue seen here (data/index inconsistency).
* #9185 fixed some issues related to lots of small deletes and compactions
* #9173 fixed regression in master introduced in ca40c1a
* #9179 fixed a regression introduced in #9084 

One of the fixes from #9185 has been back ported to 1.3 (#9186) and 1.4 (#9187) that may be one cause of deletes removing too much data.  The other fixes are specific to issues that exist in master due to #9084.  #9084 can't be back ported so upgrading to the current nightly or 1.5 when released is the only option currently if this issue still exists in 1.3 and 1.4.

**Comment by hugovalente-mindera** [comment]
@jwilder @rbetts Can you please confirm if this fix has been merged in any version of 1.5.X? 
Thanks

**Comment by e-dard** [comment]
These should all have made it into `1.5.0`
